netweaver/monitoring.sls

```diff
@@ -1,26 +1,36 @@
 {%- from "netweaver/map.jinja" import netweaver with context -%}
 {% set host = grains['host'] %}
 
+
 prometheus_sap_host_exporter_pkg:
   pkg.installed:
     - name: prometheus-sap_host_exporter
 
+# the sid, instance number pair of a node is unique, so we need to adapt configuration
+{% for node in netweaver.nodes if host == node.host %}
+
+{% set instance = node.instance %}
+{% set instance = '{:0>2}'.format(node.instance) %}
+{% set daemon_instance = '{}_{}'.format(node.sid, instance) %}
+
+# don't install exporter for db instance
+{%- if not node.sap_instance == "db" %}
```

Comment:
I would put this in the `for` loop declaration, to avoid just this if check.
Something like
```
{% for node in netweaver.nodes if host == node.host and node.sap_instance != "db" %}
```